SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs

```diff
@@ -200,31 +203,73 @@ private Task ProcessPendingErrorsAsync()
 
         private Task SendCrashReportsOrAwaitUserConfirmationAsync()
         {
-            return HandleUserConfirmationAsync();
+            bool alwaysSend = ApplicationSettings.GetValue(PrefKeyAlwaysSend, false);
+
+            return Task.Run(async () =>
+            {
+                if (_unprocessedManagedErrorLogs.Count() > 0)
+                {
+                    // Check for always send: this bypasses user confirmation callback.
+                    if (alwaysSend)
+                    {
+                        AppCenterLog.Debug(LogTag, "The flag for user confirmation is set to AlwaysSend, will send logs.");
+                        await HandleUserConfirmationAsync(UserConfirmation.Send);
+                        return;
+                    }
+
+                    if (ShouldAwaitUserConfirmation?.Invoke() ?? false)
+                    {
+                        AppCenterLog.Debug(LogTag, "ShouldAwaitUserConfirmation returned true, wait sending logs.");
+                    }
+                    else
+                    {
+                        AppCenterLog.Debug(LogTag, "ShouldAwaitUserConfirmation returned false, will send logs.");
+                        await HandleUserConfirmationAsync(UserConfirmation.Send);
+                    }
+                }
+            });
         }
 
-        private Task HandleUserConfirmationAsync()
+        private Task HandleUserConfirmationAsync(UserConfirmation userConfirmation)
         {
-            // Send every pending log.
             var keys = _unprocessedManagedErrorLogs.Keys.ToList();
             var tasks = new List<Task>();
-            foreach (var key in keys)
+
+            if (userConfirmation == UserConfirmation.DontSend)
             {
-                var log = _unprocessedManagedErrorLogs[key];
-                tasks.Add(Channel.EnqueueAsync(log));
-                _unprocessedManagedErrorLogs.Remove(key);
-                ErrorLogHelper.RemoveStoredErrorLogFile(key);
-                var errorReport = new ErrorReport(log, null);
-
-                // This must never called while a lock is held.
-                var attachments = GetErrorAttachments?.Invoke(errorReport);
-                if (attachments == null)
+                foreach (var key in keys)
```

Comment:
Instead of looping here, why not call the method to remove all error log files? And call `_unprocessedManagedErrorLogs.Clear();`?

Reply:
Also, is manipulation of `_unprocessedManagedErrorLogs` thread safe here?

Reply:
Was discussed in an earlier PR: https://github.com/microsoft/appcenter-sdk-dotnet/pull/993/files